src/Compilers/Core/Portable/Compilation/TypeInfo.cs

```diff
@@ -17,12 +17,16 @@ public struct TypeInfo : IEquatable<TypeInfo>
         /// </summary>
         public ITypeSymbol Type { get; }
 
+        public Nullability TypeNullability { get; }
+
         /// <summary>
         /// The type of the expression after it has undergone an implicit conversion. If the type
         /// did not undergo an implicit conversion, returns the same as Type.
         /// </summary>
         public ITypeSymbol ConvertedType { get; }
 
+        public Nullability ConvertedTypeNullability { get; }
```

Comment:
The only way this could be different from `TypeNullability` is if it is a user-defined conversion. In all other cases whether the expression is null or not doesn't change.

Reply:
I could see calling `GetTypeInfo` on an expression with a `!` expressing the change in nullability via this field.

---
In reply to: [205845726](https://github.com/dotnet/roslyn/pull/28860#discussion_r205845726) [](ancestors = 205845726)